simtools/configuration/commandline_parser.py

```diff
@@ -295,102 +303,126 @@ def _initialize_simulation_software(self):
             default="corsika_simtel",
         )
 
-    def _initialize_simulation_configuration(self):
-        """Initialize simulation configuration arguments."""
-        _configuration_group = self.add_argument_group("simulation configuration")
-        _configuration_group.add_argument(
-            "--primary",
-            help=(
-                "Primary particle to simulate. "
-                f"(choices for common names: {', '.join(PrimaryParticle.particle_names().keys())}; "
-                "use '--primary_ID_type' to use other particle ID types)."
-            ),
-            type=str.lower,
-            required=True,
-        )
-        _configuration_group.add_argument(
-            "--primary_id_type",
-            help="Primary particle ID type",
-            type=str,
-            required=False,
-            choices=["common_name", "corsika7_id", "pdg_id"],
-            default="common_name",
-        )
-        _configuration_group.add_argument(
-            "--azimuth_angle",
-            help=(
-                "Telescope pointing direction in azimuth. "
-                "It can be in degrees between 0 and 360 or one of north, south, east or west "
-                "North is 0 degrees and the azimuth grows clockwise (East is 90 degrees)."
-            ),
-            type=CommandLineParser.azimuth_angle,
-            required=True,
-        )
-        _configuration_group.add_argument(
-            "--zenith_angle",
-            help="Zenith angle in degrees (between 0 and 180).",
-            type=CommandLineParser.zenith_angle,
-            required=True,
-        )
-        _configuration_group.add_argument(
-            "--nshow",
-            help="Number of showers per run to simulate.",
-            type=int,
-            required=False,
-        )
-        _configuration_group.add_argument(
-            "--run_number_start",
-            help="Run number for the first run.",
-            type=int,
-            required=True,
-            default=1,
-        )
-        _configuration_group.add_argument(
-            "--number_of_runs",
-            help="Number of runs to be simulated.",
-            type=int,
-            required=True,
-            default=1,
-        )
-        _configuration_group.add_argument(
-            "--event_number_first_shower",
-            help="Event number of first shower",
-            type=int,
-            required=False,
-            default=1,
-        )
+    @staticmethod
+    def _get_dictionary_with_corsika_configuration():
+        """Return dictionary with CORSIKA configuration parameters."""
+        return {
```

Comment:
I would suggest to keep the previous way of defining the dictionary to have this consistent in the code. Defining a dictionary manually with all the entries here seems more error prone.

Reply:
I have introduced the dictionaries to allow for the simple configuration of all parameters with the `all` flag - then we simply loop over all parameters and add all of them to argpars. If we don't need all, we only configure those given in the list of input parameters.

I think this solution avoids a lot of if statements. Do you have another suggestion on how to do this?

Reply:
Wouldn't it be possible to get all added parameters from `_configuration_group`?

Reply:
Not sure I get it - `configuration_group = self.add_argument_group(group_name)` is when it is defined an empty group. We have to add arguments depending on the request in the function call.

Reply:
Yes, so the same what we did previously with `_configuration_group.add_argument(`. After this can we access the added parameters to this group?

Reply:
After talking, we decided now to keep it this way.